dask_image/ndmeasure/__init__.py

```diff
@@ -649,3 +649,97 @@ def variance(input, labels=None, index=None):
     )
 
     return var_lbl
+
+# this version of a watershed alg. is adapted from scipy.ndimage.measurement
+def watershed_ift(input, markers, structure=None, output=None):
+    """
+    Apply watershed from markers using image foresting transform algorithm.
+
+    Parameters
+    ----------
+    input : array_like
+        Input.
+    markers : array_like
+        Markers are points within each watershed that form the beginning
+        of the process.  Negative markers are considered background markers
+        which are processed after the other markers.
+    structure : structure element, optional
+        A structuring element defining the connectivity of the object can be
+        provided. If None, an element is generated with a squared
+        connectivity equal to one.
+    output : ndarray, optional
+        An output array can optionally be provided.  The same shape as input.
+
+    Returns
+    -------
+    watershed_ift : ndarray
+        Output.  Same shape as `input`.
+
+    References
+    ----------
+    .. [1] A.X. Falcao, J. Stolfi and R. de Alencar Lotufo, "The image
+           foresting transform: theory, algorithms, and applications",
+           Pattern Analysis and Machine Intelligence, vol. 26, pp. 19-29, 2004.
+
+    """
+    print("Warning: watershed is a stub")
+    warn("Warning: watershed is a stub", RuntimeWarning)
+
+    input = dask.array.asarray(input)
+
+    if input.dtype.type not in [numpy.uint8, numpy.uint16]:
+        raise TypeError('only 8 and 16 unsigned inputs are supported')
+
+    if not all([len(c) == 1 for c in input.chunks]):
+        warn("``input`` does not have 1 chunk in all dimensions; it will be consolidated first", RuntimeWarning)
+
+    if structure is None:
+        structure = scipy.ndimage.generate_binary_structure(input.ndim, 1)
+    structure = dask.array.asarray(structure, dtype=bool)
+    if structure.ndim != input.ndim:
+        raise RuntimeError('structure and input must have equal rank')
+    for ii in structure.shape:
+        if ii != 3:
+            raise RuntimeError('structure dimensions must be equal to 3')
+
+    return None # 
```

Comment:
It's probably still too early for comments on this yet, but I wanted to flag that the indentation level here suggests that this function must return None *every* time. I imagine that's not quite what's intended, unless I'm missing context.